File: app/src/main/java/com/ed/redditapp/lib/api/RedditApi.java
package com.ed.redditapp.lib.api;

import com.ed.redditapp.ui.postlist.Post;

public interface RedditApi {
    SubReddit[] searchSubreddits(String query);
    SubReddit getSubredditInfo(String subredditName);
    Post[] getSubredditPosts(String subredditName);
    String getSubredditIconUrl(String subreddit);
    Comment[] getPostComments(String postUrl);
}

File: app/src/main/java/com/ed/redditapp/lib/api/RedditApi.kt
package com.ed.redditapp.lib.api

interface RedditApi {
    fun searchSubreddits(query: String): Array<SearchItemSubreddit>
    fun getSubredditInfo(subredditName: String): SubReddit
    fun getSubredditPosts(subredditName: String): Array<Post>
    fun getSubredditIconUrl(subreddit: String): String
    fun getPostComments(postUrl: String): Array<Comment>
}